Add a warning message to retire ProcessGroup RPC backend

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #55843 [Don't Review] Test Only
* **#55616 Add a warning message to retire ProcessGroup RPC backend**

Differential Revision: [D27650627](https://our.internmc.facebook.com/intern/diff/D27650627)

mrshenli committed:
Add a warning message to retire ProcessGroup RPC backend

[ghstack-poisoned]

rohan-varma reviewed (approved):
LGTM!

pritamdamania87 commented on torch/distributed/rpc/__init__.py:
```diff
@@ -156,7 +156,10 @@ def init_rpc(
             logger.warning(
                 "RPC was initialized with the PROCESS_GROUP backend which is "
                 "deprecated and slated to be removed and superseded by the TENSORPIPE "
-                "backend. It is recommended to migrate to the TENSORPIPE backend."
+                "backend. It is recommended to migrate to the TENSORPIPE backend. "
+                "PyTorch v1.9 will be the last release that carries PROCESS_GROUP "
+                "RPC backend. If you have concerns or suggestions please comment in "
+                "https://github.com/pytorch/pytorch/issues/55615"
```
Just noticed this is a `logger.warning`, shouldn't we use python warnings instead: `warnings.warn(..., DeprecationWarning)`? This seems like the standard practice for deprecation warnings.

mrshenli replied:
good point, let me update.

mrshenli committed:
Update on "Add a warning message to retire ProcessGroup RPC backend"


Differential Revision: [D27650627](https://our.internmc.facebook.com/intern/diff/D27650627)

[ghstack-poisoned]

mrshenli committed:
Update on "Add a warning message to retire ProcessGroup RPC backend"


Differential Revision: [D27650627](https://our.internmc.facebook.com/intern/diff/D27650627)

[ghstack-poisoned]

mrshenli committed:
Update on "Add a warning message to retire ProcessGroup RPC backend"


Differential Revision: [D27650627](https://our.internmc.facebook.com/intern/diff/D27650627)

[ghstack-poisoned]
